### PR TITLE
Incremental data

### PR DESCRIPTION
**Pull request by leelanakka**

### CODE REVIEW

File: src/main/java/in/org/projecteka/hiu/consent/ConsentRepository.java [hunk 1]
```diff
@@ -290,6 +290,24 @@
...
+                                return;
+                            }
+                            var iterator = handler.result().iterator();
+                            if (!iterator.hasNext()) {
+                                monoSink.success();
+                                return;
+                            }
+                            monoSink.success(ConsentStatus.valueOf(iterator.next().getString("status")));
+                        }));
+    }
```
Comment (dheeraj-p) [inline src/main/java/in/org/projecteka/hiu/consent/ConsentRepository.java]: method name should be `consentRequetStatus`, right?

Comment (leelanakka) [reply src/main/java/in/org/projecteka/hiu/consent/ConsentRepository.java]: yea, Will change

File: src/main/java/in/org/projecteka/hiu/consent/ConsentService.java [hunk 2]
```diff
@@ -114,32 +117,89 @@ public ConsentService(HiuProperties hiuProperties,
...
+                    }
+                    for (Map<String, Object> consent : consentData) {
+                        var consentArtefactId = consent.get("consentArtefactId").toString();
+                        var consentCreatedDate = (LocalDateTime) consent.get("dateCreated");
+                        if (consentCreatedDate.isAfter(now.minusMinutes(consentServiceProperties.getConsentRequestDelay()))) {
+                            return Mono.empty();
+                        }
+                        if (consentArtefactId == null) {
+                            continue;
+                        } else {
```
Comment (dheeraj-p) [inline src/main/java/in/org/projecteka/hiu/consent/ConsentService.java]: Don't need else after a `continue`

File: src/main/java/in/org/projecteka/hiu/consent/ConsentService.java [hunk 3]
```diff
@@ -114,32 +117,89 @@ public ConsentService(HiuProperties hiuProperties,
...
+                                                    buildConsentRequest(requesterId, hipId, dateRange.getFrom());
+
+                                        }
+                                    }
+                                }
+                                return Mono.empty();
+                            });
+                        }
+                    }
+                    return Mono.empty();
```
Comment (dheeraj-p) [inline src/main/java/in/org/projecteka/hiu/consent/ConsentService.java]: Can this be extracted out into a method?

File: src/main/java/in/org/projecteka/hiu/consent/ConsentService.java (hunk 2 above)
Comment (leelanakka) [reply src/main/java/in/org/projecteka/hiu/consent/ConsentService.java]: done

File: src/main/java/in/org/projecteka/hiu/consent/ConsentService.java (hunk 3 above)
Comment (leelanakka) [reply src/main/java/in/org/projecteka/hiu/consent/ConsentService.java]: Tried but no luck, Need to pass lot of context to function

**Review by dheeraj-p — APPROVED** [review]

**Pull request merged**